stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/persistence/OfflinePlugin.kt

```diff
@@ -0,0 +1,24 @@
+package io.getstream.chat.android.client.experimental.persistence
+
+import io.getstream.chat.android.client.experimental.plugin.Plugin
+import io.getstream.chat.android.client.experimental.plugin.listeners.ChannelMarkReadListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.EditMessageListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.GetMessageListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.HideChannelListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.MarkAllReadListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.QueryChannelListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.QueryChannelsListener
+import io.getstream.chat.android.client.experimental.plugin.listeners.ThreadQueryListener
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+
+@ExperimentalStreamChatApi
+public interface OfflinePlugin :
```

Comment:
Now the `OfflinePlugin`  doesn't violate the Open/Close principle anymore. If we need to change/extend the behavior of `OfflinePlugin`, we can change the implementation of one of its dependencies, but we don't need to modify the code of the `OfflinePlugin`. This way, changes happen in much more specific and smaller places, which makes changes easier to reason about. 

Dependency inversion is also respected. Now the class doesn't instantiate its own dependencies and just receive the interfaces that if needs to perform its job.

Reply:
This interface is not necessary 🤦 . But the documentation applies to the OfflinePluginImpl as well.